mmv1/third_party/terraform/resources/resource_compute_security_policy.go.erb

```diff
@@ -768,6 +806,31 @@ func flattenMatchExpr(match *compute.SecurityPolicyRuleMatcher) []map[string]int
 	return []map[string]interface{}{data}
 }
 
+func expandSecurityPolicyAdvancedOptionsConfig(configured []interface{}) *compute.SecurityPolicyAdvancedOptionsConfig {
+	if len(configured) == 0 || configured[0] == nil {
+		return nil
```

Comment:
Ah, yes, ok, so the drawback to this is if `unset` differs from `default`, which it sounds like it does in this case. This will assume that unset will ALWAYS default.

There shouldn't be a diff, I don't think? Because the state will match the default. You might have to make the parent (`advanced_options_config`) `Optional` and `Computed` maybe, but I'd have to do some playing around with that to confirm it. However, an empty configuration, with defaults, should not conflict with defaults in state. It's possible I'm missing something here, as usually I like to test something out - I just don't have that time at the current moment, but I can mess with it more later perhaps.

The other option would be to make them `Optional` and `Computed`, but then we're going to still hit the upstream bug, I'd imagine - it would probably just get rid of the diff (which, I suppose is fine - we'd be mimicking the API at least).

Would you mind trying those options? If they both work, pick which solution you like best knowing the drawbacks.

Reply:
Ok, update! They both pass, however, trying to hardcode the default doesn't work as intended (i think it's a long known SDK issue that i still haven't figured out how to work around), so I think our best solution would be to make everything (`advanced_options_config` and its children) `Optional` and `Computed`. This will mimic the API so that the bug will still be present, and removing the configuration won't do anything, but our permadiff will go away. The only way to reset it to the defaults would be to update the configuration to the appropriate value. Does this work for you?

Reply:
> Ok, update! They both pass,

Thanks for testing it - really appreciate it.

> The only way to reset it to the defaults would be to update the configuration to the appropriate value. Does this work for you?

If I'm understanding you right, yes, I think that works. To summarize / paraphrase, I think the behavior is
* defaults / advanced options _not_ set, creating new resource or updating provider for existing resource: no changes
* options set explicitly away from (or to) default: options will be set as requested, however, once removed, the configuration would _then_ need to have (both) the default values (or whatever values they want) explicitly set in the config.

Reply:
Correct!

Reply:
thanks. Yes, I think that's the best we can do.

@megan07 I rebased, added `Computed: true,` to both the parent and children, and pushed up an update. I have not yet tested this locally. Do you want to confirm that this is what you tested and what you were suggesting?

Reply:
I also had to remove the defaults - otherwise, failed with this error:
```
│ 1 error occurred:
│ 	* resource google_compute_security_policy: json_parsing: Default must be nil if computed
```

Reply:
Commented below in the main comments with how this behaves with some simple functional tests.